stl/inc/ranges

```diff
@@ -703,6 +703,10 @@ namespace ranges {
             return *this;
         }
 
+        _NODISCARD constexpr explicit operator bool() const noexcept requires _Needs_operator_bool {

```

Comment:
Do we really need the requires clause?

Reply:
It's technically not needed but it makes sure that we never accidentally use this function without explicitly requesting it (which might cause bugs in the future)

Reply:
+1 programs that check `operator bool` without explicitly requesting it should fail to compile.